types/index.d.ts

```diff
@@ -0,0 +1,369 @@
+// TypeScript Version: 3.5
+
+/*
+ * Created by Dan Freeman
+ * https://github.com/dfreeman/
+ *
+ * Source: https://gist.github.com/dfreeman/33fc80164c0ad91d5e9480a94aa6454c#file-tests-model-ts
+ */
+declare module "miragejs" {
+  import {
+    FactoryDefinition,
+    ModelDefinition,
+    ModelInstance
+  } from "miragejs/-types";
+  export { Server } from "miragejs/server";
+  export { Registry, ModelInstance } from "miragejs/-types";
+
+  /**
+   * A fake HTTP request
+   */
+  export class Request {
```

Comment:
Can't we reuse the `Request` from pretender? https://github.com/pretenderjs/pretender/blob/master/index.d.ts#L47

Reply:
There should be more reusable stuff too

Reply:
I'm not sure how could you reuse, any idea?

Reply:
Oh, I thought `pretender` was exporting the FakeXMLHttpRequest type, which it is not :( Never mind then

Reply:
Maybe there is some way to reuse them, it would be great. Let's figure it out later. ;) Thanks for the suggestion. Definitely an important area where we can improve the type def further.